tests/bluetooth/host/keys/bt_keys_add_type/src/main.c

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/zephyr.h>
+#include <bluetooth/bluetooth.h>
+#include "testing_common_defs.h"
+#include <keys.h>
+#include "host_mocks/assert.h"
+#include "mocks/rpa.h"
+#include "mocks/keys_help_utils.h"
+
+/* This LUT contains different combinations of ID, Address and key type.
+ * Item in this list will be used to fill keys pool.
+ */
+static const struct id_addr_type testing_id_addr_type_lut[CONFIG_BT_MAX_PAIRED] = {
+
+	{ BT_ADDR_ID_1, BT_ADDR_LE_1, BT_KEYS_PERIPH_LTK },
+	{ BT_ADDR_ID_1, BT_RPA_ADDR_LE_1, BT_KEYS_PERIPH_LTK },
+	{ BT_ADDR_ID_1, BT_RPA_ADDR_LE_2, BT_KEYS_IRK },
```

Comment:
I'll dig into the implications more when I get to the use of this list, but strictly speaking it's an error to have an IRK without a LTK or a CSRK without an LTK, so if these are populating a key-pool record, it may be testing the validity of an invalid case which may be fixed in future.

Reply:
Actually, I don't know much about these IRK, LTK and CSRK and the relations between them.
The implementation of `bt_keys_add_type` is pretty simple, it just sets what you pass to it.

I'd say that in a Test-Driven-Development, the test case should be designed to consider such relations.
I'm not sure if I should be restricted to the current implementation while designing test case, or I should go beyond the implementation.
Also, you were able to catch such thing because you have the right knowledge, but I didn't know about it.